kubechain/api/v1alpha1/mcpserver_types.go

```diff
@@ -0,0 +1,141 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// MCPServerSpec defines the desired state of MCPServer
+type MCPServerSpec struct {
+	// Transport specifies the transport type for the MCP server
+	// +kubebuilder:validation:Enum=stdio;http
+	// +kubebuilder:validation:Required
+	Transport string `json:"transport"`
+
+	// Command is the command to run for stdio MCP servers
+	// +optional
+	Command string `json:"command,omitempty"`
+
+	// Args are the arguments to pass to the command for stdio MCP servers
+	// +optional
+	Args []string `json:"args,omitempty"`
+
+	// Env are environment variables to set for stdio MCP servers
+	// +optional
+	Env []EnvVar `json:"env,omitempty"`
+
+	// URL is the endpoint for HTTP MCP servers
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// ResourceRequirements defines CPU/Memory resources requests/limits
+	// +optional
+	Resources ResourceRequirements `json:"resources,omitempty"`
+}
+
+// EnvVar represents an environment variable
+type EnvVar struct {
+	// Name of the environment variable
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Value of the environment variable (direct literal value)
+	// +optional
+	Value string `json:"value,omitempty"`
+
+	// ValueFrom represents a source for the value of an environment variable
+	// +optional
+	ValueFrom *EnvVarSource `json:"valueFrom,omitempty"`
```

Comment:
probably overkill for now, but did you consider importing from upstream? https://github.com/kubernetes/kubernetes/blob/9a8c2a9fa362db2ba59aa56481772060073c99d5/pkg/apis/core/types.go#L2203

on the other hand, i kinda like just copying here since it means we can articulate in the schema that we only support a subset of the "standard" pod spec methods for env stuff